### ISSUE TITLE
DPI scaling is incorrectly determined on X11 Linux

### ISSUE THREAD

**Issue by whitequark** [ngscopeclient/scopehal-apps#724]
This happens because it uses GLFW and GLFW currently retrieves content scale via "folklore heuristics" (their words) which do not really work: https://github.com/glfw/glfw/blob/b35641f4a3c62aa86a0b3c983d163bc0fe36026d/src/x11_init.c#L990-L1023

On X11 there's no unified way to do this. The best you could do in a fully arch-independent manner is to query RandR display DPI, but that's kind of a pain to do. An alternative is to look at GDK/Qt environment variables.

**Comment by whitequark** [comment]
I've submitted https://github.com/ngscopeclient/scopehal-apps/pull/725 as an example of how to make DPI scaling determination work pretty okay on X11 Linux, but it still needs to be tested on Wayland Linux, Windows, and macOS.

**Comment by nemanjan00** [comment]
Scaling also seems to be completely broken on Wayland. 

I am not sure if cause is also GLFW

![20240701_13h47m09s_grim](https://github.com/ngscopeclient/scopehal-apps/assets/448151/26bcb074-9bc8-4f50-a9cb-fe0048a50f00)
![20240701_13h47m25s_grim](https://github.com/ngscopeclient/scopehal-apps/assets/448151/2029195c-6309-4ff0-9faf-b1deedf09ef5)

I have a hack (temporary "solution")  that might help some people. 

It does seem to at least fix cursor alignment to buttons and to render with reasonable resolution

```
gamescope -W 1440 -H 900 -r 60 -- ngscopeclient
```

![20240701_13h53m47s_grim](https://github.com/ngscopeclient/scopehal-apps/assets/448151/8e9e4fb7-ed8c-4604-801a-bbe406ed22bd)

**Comment by MonarchActual** [comment]
I have the same issue on wayland (sway).
`gamescope` also does not work because of an issue with my intel integrated GPU.
As a workaround I use Xwayland using `XDG_SESSION_TYPE=x11`.
This works fine for me but I am using the default scale of 1 / DPI of 96.

**Comment by whitequark** [comment]
I will have to test this, I'll do it at the same time as #1022.

**Comment by whitequark** [comment]
Nope, ngscopeclient is still unusably tiny on KDE/X11 in revision 0dc295532fdb40fc84d3c0298b791ca121546209

**Comment by whitequark** [comment]
The cause is likely that Xft.dpi is 96:

```
$ xrdb -query
*customization: -color
Xcursor.size:   48
Xcursor.theme:  breeze_cursors
Xft.antialias:  1
Xft.dpi:        96
Xft.hinting:    0
Xft.hintstyle:  hintnone
Xft.rgba:       none
```

I cannot set it to a larger value because then the scaling gets applied twice and instead of ngscopeclient being unusably tiny, every KDE (or Electron, I don't remember which of them broke) application becomes unusably large.
